XenCenterLib/StringUtility.cs

```diff
@@ -224,19 +226,32 @@ public static bool IsIPAddress(string s)
         /// Prepends \\?\ to path if it is longer than (MAX_PATH - 12) in Windows. This is the legacy directory length limit.<br /><br />
         /// See <see href='https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilea'>CreateFileA</see> and <see href='https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry'>Maximum Path Length Limitation</see> for more info.
         /// </summary>
-        /// <param name="inputPath">The input file path</param>
-        /// <param name="force">Set to true to prepend \\?\ to the path regardless of input length</param>
-        /// <returns></returns>
-        public static string ToLongWindowsPath(string inputPath, bool force = false)
+        /// <param name="inputPath">The input path</param>
+        /// <param name="isDirectory">Whether the input path is a directory</param>
+        public static string ToLongWindowsPath(string inputPath, bool isDirectory)
         {
-            if (inputPath.StartsWith(@"\\?\"))
-            {
+            if (string.IsNullOrEmpty(inputPath) || inputPath.StartsWith(@"\\?\"))
                 return inputPath;
-            }
-            if (!force && (string.IsNullOrEmpty(inputPath) || inputPath.Length < 248))
-            {
+
+            if (isDirectory)
+                return inputPath.Length >= 248 ? ToLongWindowsPathUnchecked(inputPath) : inputPath;
+            
+            var dir = Path.GetDirectoryName(inputPath);

```

Comment:
can't `Path.GetDirectoryName` fail with `inputPath >= 260`?

Reply:
never mind I checked and it works without it, looks like that method doesn't hit the length limit

Reply:
From what I understood, the path operations do not fail when they're "theoretical" like in this case where it's splitting the path, but rather when they actually access (read or write) the file system (I hope I have understood it correctly 😨).